tardis/adapters/sites/lancium.py

```diff
@@ -0,0 +1,141 @@
+        self._machine_type = machine_type
+        self._site_name = site_name
+
+        auth = Authenticator(api_key=self.configuration.api_key)
+        self.client = LanciumClient(api_url=self.configuration.api_url, auth=auth)
+
+        key_translator = StaticMapping(
+            remote_resource_uuid="id",
+            drone_uuid="name",
+            resource_status="status",
+        )
+
+        translator_functions = StaticMapping(
+            status=lambda x, translator=StaticMapping(
+                **self.resource_status_translation
+            ): translator[x],
+            id=int,
+            name=str,
+        )
+
+        self.handle_response = partial(
+            self.handle_response,
+            key_translator=key_translator,
+            translator_functions=translator_functions,
+        )
+
+        self._lancium_status = AsyncCacheMap(
+            update_coroutine=partial(lancium_status_updater, self.client),
+            max_age=self.configuration.max_age * 60,
+        )
+
+    async def deploy_resource(
+        self, resource_attributes: AttributeDict
+    ) -> AttributeDict:
+        specs = dict(name=resource_attributes.drone_uuid)
+        specs["resources"] = dict(
+            core_count=self.machine_meta_data.Cores,
+            memory=self.machine_meta_data.Memory,
+            scratch=self.machine_meta_data.Disk,
+        )
+        specs["environment"] = [
+            {"variable": f"TardisDrone{key}", "value": str(value)}
+            for key, value in self.drone_environment(
+                resource_attributes.drone_uuid,
+                resource_attributes.obs_machine_meta_data_translation_mapping,
+            ).items()
+        ]
+        specs.update(self.machine_type_configuration)
+        create_response = await self.client.jobs.create_job(job=specs)
+        logger.debug(f"{self.site_name} create job returned {create_response}")
+        submit_response = await self.client.jobs.submit_job(
+            id=create_response["job"]["id"]
+        )
+        logger.debug(f"{self.site_name} submit job returned {submit_response}")
+        return self.handle_response(create_response["job"])
+
+    async def resource_status(
+        self, resource_attributes: AttributeDict
+    ) -> AttributeDict:
+        await self._lancium_status.update_status()
+        # In case the created timestamp is after last update timestamp of the
+        # asynccachemap, no decision about the current state can be given,
+        # since map is updated asynchronously.
+        try:
+            resource_uuid = resource_attributes.remote_resource_uuid
+            resource_status = self._lancium_status[resource_uuid]
+        except KeyError as err:
+            if (
+                self._lancium_status._last_update - resource_attributes.created
+            ).total_seconds() < 0:
+                raise TardisResourceStatusUpdateFailed from err
+            else:
```

Comment:
I don't exactly get the else case.
Why the state is set to deleted then?

Reply:
Okay. We are caching the `show_jobs` call for a configurable time interval. If a job starts up, it can happen that it is not already in the cached list of jobs. So, if the time difference between the last update and the job creation is negative, we will retry to get the status of the jobs by raising `TardisResourceStatusUpdateFailed`. 

However, once a job is finished and has been deleted, it will not show up in the `show_jobs` call as well. The time difference between the last update and the job creation is positive, so the status is set to deleted in that case.

Reply:
Ah okay, I got it, thanks 👍